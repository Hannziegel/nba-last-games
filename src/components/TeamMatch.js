import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Teams.css';

const TeamMatch = (props) => {
  const {
    teamLastGames,
  } = props;

  return (
    <div
      className="team-match-thumb"
      style={{ backgroundImage: `url(${teamLastGames.strThumb})` }}
    >
      <div className="team-match-container">
        <div className="home-team">
          <div className="match-team-info">
            <h4 className="home-team-name">{teamLastGames.strHomeTeam}</h4>
            <p className="home-team-score">{teamLastGames.intHomeScore}</p>
          </div>
        </div>
        <div className="vs">VS.</div>
        <div className="away-team">
          <div className="match-team-info">
            <h4 className="away-team-name">{teamLastGames.strAwayTeam}</h4>
            <p className="away-team-score">{teamLastGames.intAwayScore}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

TeamMatch.propTypes = {
  teamLastGames: PropTypes.shape({
    strHomeTeam: PropTypes.string,
    intHomeScore: PropTypes.string,
    strAwayTeam: PropTypes.string,
    intAwayScore: PropTypes.string,
    strThumb: PropTypes.string,
  }).isRequired,
};

export default TeamMatch;
