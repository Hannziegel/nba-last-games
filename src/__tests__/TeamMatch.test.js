import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../Redux/configureStore'; // eslint-disable-line
import TeamMatch from '../components/TeamMatch';

describe('App js testing', () => {
  const teamLastGames = {
    strHomeTeam: 'PropTypes.string',
    intHomeScore: 'PropTypes.string',
    strAwayTeam: 'PropTypes.string',
    intAwayScore: 'PropTypes.string',
    strThumb: 'PropTypes.string',
  };

  test('Renders App', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <TeamMatch teamLastGames={teamLastGames} />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
