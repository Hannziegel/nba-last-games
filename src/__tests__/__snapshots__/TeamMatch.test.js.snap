// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App js testing Renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="team-match-thumb"
        style="background-image: url(PropTypes.string);"
      >
        <div
          class="team-match-container"
        >
          <div
            class="home-team"
          >
            <div
              class="match-team-info"
            >
              <h4
                class="home-team-name"
              >
                PropTypes.string
              </h4>
              <p
                class="home-team-score"
              >
                PropTypes.string
              </p>
            </div>
          </div>
          <div
            class="vs"
          >
            VS.
          </div>
          <div
            class="away-team"
          >
            <div
              class="match-team-info"
            >
              <h4
                class="away-team-name"
              >
                PropTypes.string
              </h4>
              <p
                class="away-team-score"
              >
                PropTypes.string
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="team-match-thumb"
      style="background-image: url(PropTypes.string);"
    >
      <div
        class="team-match-container"
      >
        <div
          class="home-team"
        >
          <div
            class="match-team-info"
          >
            <h4
              class="home-team-name"
            >
              PropTypes.string
            </h4>
            <p
              class="home-team-score"
            >
              PropTypes.string
            </p>
          </div>
        </div>
        <div
          class="vs"
        >
          VS.
        </div>
        <div
          class="away-team"
        >
          <div
            class="match-team-info"
          >
            <h4
              class="away-team-name"
            >
              PropTypes.string
            </h4>
            <p
              class="away-team-score"
            >
              PropTypes.string
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
